packages/playwright/src/executors/playwright/playwright.ts

```diff
@@ -0,0 +1,183 @@
+    );
+  }
+  const args = createArgs(options);
+  const p = runPlaywright(args, projectRoot);
+
+  return new Promise<{ success: boolean }>((resolve) => {
+    // TODO(caleb): are there other signals I need to care about?
+    p.on('close', (code) => {
+      resolve({ success: code === 0 });
+    });
+  });
+}
+
+function createArgs(opts: PlaywrightExecutorSchema): string[] {
+  //NOTE: boolean options send at --arg=false might be invalid options.
+  const args: string[] = [];
+  if (opts.browser !== undefined) {
+    args.push(`--browser=${opts.browser}`);
+  }
+  if (opts.config !== undefined) {
+    args.push(`--config=${opts.config}`);
+  }
+  if (opts.debug !== undefined) {
+    args.push(`--debug`);
+  }
+  if (opts.forbidOnly) {
+    args.push(`--forbid-only`);
+  }
+  if (opts.fullyParallel) {
+    args.push(`--fully-parallel`);
+  }
+  if (opts.grep !== undefined) {
+    args.push(`--grep=${opts.grep}`);
+  }
+  if (opts.globalTimeout !== undefined) {
+    args.push(`--global-timeout=${opts.globalTimeout}`);
+  }
+  if (opts.grepInvert !== undefined) {
+    args.push(`--grep-invert=${opts.grepInvert}`);
+  }
+  if (opts.headed) {
+    args.push(`--headed`);
+  }
+  if (opts.ignoreSnapshots) {
+    args.push(`--ignore-snapshots`);
+  }
+  if (opts.workers !== undefined) {
+    args.push(`--workers=${opts.workers}`);
+  }
+  if (opts.list) {
+    args.push(`--list`);
+  }
+  if (opts.maxFailures !== undefined) {
+    args.push(
+      typeof opts.maxFailures === 'number'
+        ? `--max-failures=${opts.maxFailures}`
+        : `-x`
+    );
+  }
+  if (opts.noDeps) {
+    args.push(`--no-deps`);
+  }
+  if (opts.output !== undefined) {
+    args.push(`--output=${opts.output}`);
+  }
+  if (opts.passWithNoTests) {
+    args.push(`--pass-with-no-tests`);
+  }
+  if (opts.project !== undefined) {
+    args.push(`--project=${opts.project.map((p) => p.trim()).join(',')}`);
+  }
+  if (opts.quiet) {
+    args.push(`--quiet`);
+  }
+  if (opts.repeatEach !== undefined) {
+    args.push(`--repeat-each=${opts.repeatEach}`);
+  }
+  if (opts.reporter !== undefined) {
+    args.push(`--reporter=${opts.reporter}`);
+  }
+  if (opts.retries !== undefined) {
+    args.push(`--retries=${opts.retries}`);
+  }
+
+  if (opts.shard !== undefined) {
+    args.push(`--shard=${opts.shard}`);
+  }
+  if (opts.timeout !== undefined) {
+    args.push(`--timeout=${opts.timeout}`);
+  }
+  if (opts.trace !== undefined) {
+    args.push(`--trace=${opts.trace}`);
+  }
+  if (opts.updateSnapshots) {
+    args.push(`--update-snapshots`);
+  }
+  if (opts.ui) {
+    args.push(`--ui`);
+  }
+  if (opts.uiHost !== undefined) {
+    args.push(`--ui-host=${opts.uiHost}`);
+  }
+  if (opts.uiPort !== undefined) {
+    args.push(`--ui-port=${opts.uiPort}`);
+  }
+
+  return args;
+}
+
+function runPlaywright(args: string[], cwd: string) {
+  const pm = getPackageManagerCommand();
+  const cmd = [pm.exec, 'playwright', 'test', ...args];
+  return spawn(cmd.join(' '), {
```

Comment:
maybe use fork instead of spawn?

Reply:
We had issues with `exec` and `spawn` not exiting the child process when the Nx process exits. Something to check here.

Like Emily mentioned, could use `fork` instead.

```
fork(require.resolve('playwright/cli'), [...]);
```